### PR TITLE
[Transform] Integration test for the case of empty key in the terms agg in transform

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR adds an integration test for the case of empty key in the terms agg in transform.

Relates https://github.com/elastic/elasticsearch/issues/90011

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/part-3

File: x-pack/plugin/transform/qa/single-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/TransformPivotRestSpecialCasesIT.java [hunk 1]
```diff
@@ -252,6 +252,99 @@ public void testRestrictiveBucketSelector() throws Exception {
...
+        {
+            final Request request = new Request("PUT", indexName + "/_doc/strange-business-id-3");
+            request.addParameter("refresh", "true");
+            request.setJsonEntity("""
+                {
+                  "user_id": "user_0",
+                  "business_id": ".business_y"
+                }""");
+            client().performRequest(request);
+        }
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/qa/single-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/TransformPivotRestSpecialCasesIT.java:287]: can you add another record with `...` as value?

**Review by hendrikmuhs — COMMENTED** [review]
LGTM, I suggest to add 1 more record to cover another corner case

**Review by hendrikmuhs — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java [hunk 2]
```diff
@@ -1859,6 +1859,18 @@ public void testBlankFieldNamesSubobjectsFalse() throws Exception {
...
+        {
+            MapperParsingException err = expectThrows(MapperParsingException.class, () -> mapper.parse(source(b -> b.field("", "foo"))));
+            assertThat(err.getMessage(), containsString("Field name cannot contain only whitespace: []"));
+        }
+        {
+            MapperParsingException err = expectThrows(MapperParsingException.class, () -> mapper.parse(source(b -> b.field("  ", "foo"))));
+            assertThat(err.getMessage(), containsString("Field name cannot contain only whitespace: [  ]"));
+        }
+    }
+
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java:1873]: Is this additional test still needed given the tests I am adding with #91043 ?

**Review by javanna — COMMENTED** [review]
the integration test looks good to me, thanks for adding it.

File: server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java (hunk 2 above)
Comment (przemekwitek) [reply server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java:1873]: No, reverted.

**Pull request merged**